x/xcvm/types/msg.go

```diff
@@ -0,0 +1,87 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	TypeMsgSendTransferIntent           = "send_transfer_intent"
+	TypeMsgVerifyTransferIntentProof    = "verify_transfer_intent_proof"
+	TypeMsgTriggerTransferIntentTimeout = "trigger_transfer_intent_timeout"
+)
+
+var _ sdk.Msg = &MsgSendTransferIntent{}
+
+// Type Implements Msg.
+func (MsgSendTransferIntent) Type() string { return TypeMsgSendTransferIntent }
+
+// GetSigners returns the expected signers for a MsgSendTransferIntent message.
+func (msg *MsgSendTransferIntent) GetSigners() []sdk.AccAddress {
+	addr, _ := sdk.AccAddressFromBech32(msg.FromAddress)
+	return []sdk.AccAddress{addr}
+}
+
+// ValidateBasic does a sanity check on the provided data.
+func (msg *MsgSendTransferIntent) ValidateBasic() error {
+	// validate from address
+	if _, err := sdk.AccAddressFromBech32(msg.FromAddress); err != nil {
+		return errorsmod.Wrap(err, "invalid from address")
+	}
+
+	// validate destination address
+	if !common.IsHexAddress(msg.DestinationAddress) {
+		return ErrInvalidDestinationAddress
+	}
+
+	// validate clientId
+	//if err := host.ClientIdentifierValidator(msg.ClientId); err != nil {
+	//	return err
+	//}
```

Comment:
necessary?

Reply:
We should keep this for basic client id validations, I'll uncomment!